marlowe-cli/src/Language/Marlowe/CLI/Export.hs

```diff
@@ -116,20 +123,15 @@ exportMarlowe :: MonadError CliError m
               -> m ()                   -- ^ Action to export the contract and transaction information to a file.
 exportMarlowe marloweParams costModel network stake contractFile stateFile inputFiles outputFile printStats =
   do
-    contract <- decodeFileStrict contractFile
-    state    <- decodeFileStrict stateFile
-    inputs   <- mapM decodeFileStrict inputFiles
-    marloweInfo@MarloweInfo{..} <-
-      liftEither
-        $ buildMarlowe
-            marloweParams costModel network stake
-            contract state
-            inputs
+    contract :: Contract <- decodeFileStrict contractFile
+    state    :: State <- decodeFileStrict stateFile
+    inputs   :: [Input] <- mapM decodeFileStrict inputFiles
+    marloweInfo@MarloweInfo{..} <- liftEither =<< asksEra \era -> buildMarlowe marloweParams era costModel network stake contract state inputs
     let
       ValidatorInfo{..} = validatorInfo
       DatumInfo{..}     = datumInfo
       RedeemerInfo{..}  = redeemerInfo
-    maybeWriteJson outputFile (marloweInfo :: MarloweInfo AlonzoEra) -- FIXME: Generalize eras.
+    doWithShelleyBasedEra $ maybeWriteJson outputFile marloweInfo
```

Comment:
Nice!

Reply:
The magic of GATDs: summoning typeclass constraints out of thin air!